src/pybind/mgr/cephadm/module.py

```diff
@@ -2041,6 +2041,7 @@ def matches_network(host):
         remove_daemon_hosts: Set[orchestrator.DaemonDescription] = ha.remove_daemon_hosts(hosts)
         self.log.debug('Hosts that will loose daemons: %s' % remove_daemon_hosts)
 
+        r = True
```

Comment:
```suggestion
        r = bool(add_daemon_hosts)
```

Let's avoid returning true, if there are no hosts to add

Reply:
Hm.  we probably also want that to be conditional on removal too, though.  Let me rethink that logic

Reply:
I've come to the conclusion I don't know what the right return code is.  Pre-existing on the osd application returns False, saying "avoid busy loop", but...the entire point is to return True if you've change the daemon complement on the cluster *in order* to cause a loop.  It seems to me the return should be True if *any* daemon was started or stopped, and only False if nothing at all was changed.  Do you agree?

Reply:
> the entire point is to return True if you've change the daemon complement on the cluster _in order_ to cause a loop. 

Yes.

> It seems to me the return should be True if _any_ daemon was started or stopped, and only False if nothing at all was changed. 

Yes.

> Do you agree? If so, that pre-existing "return False for osd addition" seems wrong.

Yes. OSDs are unfortunately a bit special.

Reply:
...but not in this regard; that return really should be True, correct?

Reply:
You should only return True, **iff** you deployed anything.